test/safari-rules.test.js

```diff
@@ -1,224 +1,9 @@
-    expect(Array.isArray(value)).toBe(true);
-    expect(value.length).not.toBeLessThan(1);
-    for (let j = 0; j < value.length; j += 1) {
-      const elem = value[j];
-      expect(typeof elem).toBe('string');
-      expect(elem === elem.toLowerCase()).toBe(true);
-      // TODO: test if string is punycode encoded
-    }
-  }
-
-  // unless-top-url must be a non-empty array of punnycode lowercase strings
-  value = rule.trigger['unless-top-url'];
-  if (value !== undefined) {
-    expect(Array.isArray(value)).toBe(true);
-    expect(value.length).not.toBeLessThan(1);
-    for (let j = 0; j < value.length; j += 1) {
-      const elem = value[j];
-      expect(typeof elem).toBe('string');
-      expect(elem === elem.toLowerCase()).toBe(true);
-      // TODO: test if string is punycode encoded
-    }
-  }
-
-  // type must be a string with the following possible values
-  value = rule.action.type;
-  expect(allowedActions.has(value)).toBe(true);
-
-  // if type is css-display-none, a non-empty selector must exist
-  if (rule.action.type === 'css-display-none') {
-    expect(rule.action.selector).not.toBeUndefined();
-    expect(rule.action.selector).not.toBe('');
-  }
-
-  // selector must be a non-empty string
-  value = rule.action.selector;
-  if (value !== undefined) {
-    expect(typeof value).toBe('string');
-    expect(value).not.toBe('');
-  }
-
-  // trigger cannot contain both if-domain and unless-domain
-  const ifdomain = rule.trigger['if-domain'];
-  const unlessdomain = rule.trigger['unless-domain'];
-  if (ifdomain !== undefined) {
-    expect(unlessdomain).toBeUndefined();
-  } else if (unlessdomain !== undefined) {
-    expect(ifdomain).toBeUndefined();
-  }
-
-  // trigger cannot contain both if-top-url and unless-top-url
-  const ifurl = rule.trigger['if-top-url'];
-  const unlessurl = rule.trigger['unless-top-url'];
-  if (ifurl !== undefined) {
-    expect(unlessurl).toBeUndefined();
-  } else if (unlessurl !== undefined) {
-    expect(ifurl).toBeUndefined();
-  }
-}
+import { convertAndValidateFilters } from '../scripts/convert-to-safari-rules';
 
 test('rule must be a valid safari rule', () => {
-
-  const { networkFilters, cosmeticFilters } = parseList(loadAllLists());
-  let nullCount = 0;
-
-  for (let i = 0; i < cosmeticFilters.length; i += 1) {
-    const cosmetic = cosmeticFilters[i];
-    const rule = convertCosmetics(cosmetic);
-    if (rule !== null && rule !== undefined) {
-      try {
-        testRule(rule);
-      } catch (ex) {
-        break;
-      }
-    } else {
-      nullCount += 1;
-    }
-  }
-
-  for (let j = 0; j < networkFilters.length; j += 1) {
-    const filter = networkFilters[j];
-    const rule = convertFilter(filter);
-    if (rule !== null && rule !== undefined) {
-      try {
-        testRule(rule);
-      } catch (ex) {
-        break;
-      }
-    } else {
-      nullCount += 1;
-    }
-  }
+  const lists = parseList(loadAllLists());
+  convertAndValidateFilters(lists);
```

Comment:
This only works if `convertAndValidateFilters` throws an exception. At the time being you catch exceptions and return an empty array.